app/code/Magento/Cms/Block/BlockByIdentifier.php

```diff
@@ -0,0 +1,170 @@
+use Magento\Framework\View\Element\Context;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * This class is replacement of \Magento\Cms\Block\Block, that accepts only `string` identifier of CMS Block
+ */
+class BlockByIdentifier extends AbstractBlock implements IdentityInterface
+{
+    public const CACHE_KEY_PREFIX = 'CMS_BLOCK';
+
+    /**
+     * @var GetBlockByIdentifierInterface
+     */
+    private $blockByIdentifier;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var FilterProvider
+     */
+    private $filterProvider;
+
+    /**
+     * @var BlockInterface
+     */
+    private $cmsBlock;
+
+    /**
+     * @param GetBlockByIdentifierInterface $blockByIdentifier
+     * @param StoreManagerInterface $storeManager
+     * @param FilterProvider $filterProvider
+     * @param Context $context
+     * @param array $data
+     */
+    public function __construct(
+        GetBlockByIdentifierInterface $blockByIdentifier,
+        StoreManagerInterface $storeManager,
+        FilterProvider $filterProvider,
+        Context $context,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->blockByIdentifier = $blockByIdentifier;
+        $this->storeManager = $storeManager;
+        $this->filterProvider = $filterProvider;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function _toHtml(): string
+    {
+        try {
+            return $this->filterOutput(
+                $this->getCmsBlock()->getContent()
+            );
+        } catch (NoSuchEntityException $e) {
+            return '';
+        }
+    }
+
+    /**
+     * Returns the value of `identifier` injected in `<block>` definition
+     *
+     * @return string|null
+     */
+    private function getIdentifier(): ?string
+    {
+        return $this->getdata('identifier') ?: null;
+    }
+
+    /**
+     * Filters the Content
+     *
+     * @param string $content
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    private function filterOutput(string $content): string
+    {
+        return $this->filterProvider->getBlockFilter()
+            ->setStoreId($this->getCurrentStoreId())
+            ->filter($content);
+    }
+
+    /**
+     * Loads the CMS block by `identifier` provided as an argument
+     *
+     * @return BlockInterface|BlockModel
+     * @throws NoSuchEntityException
+     */
+    private function getCmsBlock(): BlockInterface
+    {
+        if (!$this->getIdentifier()) {
+            throw new NoSuchEntityException(
+                __('Expected value of `identifier` was not provided')
+            );
+        }
+
+        if (null === $this->cmsBlock) {
+            $this->cmsBlock = $this->blockByIdentifier->execute(
+                (string)$this->getIdentifier(),
+                $this->getCurrentStoreId()
+            );
+        }
+
+        return $this->cmsBlock;
```

Comment:
What happens if the CMS block is disabled?

Reply:
That is actually interesting catch, as

https://github.com/magento/magento2/blob/c74256b9c637ac9808f8b0877d895b4a9b44f90f/app/code/Magento/Cms/Model/GetBlockByIdentifier.php#L43-L54

Don't care about Block status. I'm going to adjust the logic.